frontend/components/JoinGraveBtn.tsx

```diff
@@ -156,6 +156,109 @@ export default function JoinGraveBtn({
 
   // ========================= JOINING FLOW =========================
 
+  const batchJoin = async (
+    provider: ethers.providers.Web3Provider,
+    signer: ethers.providers.JsonRpcSigner
+  ): Promise<{ vaultAddress: string | null }> => {
+    const UP = new ethers.Contract(
+      account as string,
+      UniversalProfile.abi,
+      provider
+    );
+
+    // 1.create vault
+    const vaultFactory = new ethers.ContractFactory(
+      LSP9Vault.abi,
+      LSP9Vault.bytecode,
+      signer
+    );
+    const deployTransactionObject = vaultFactory.getDeployTransaction(account);
+    const firstCreateEncodedData = deployTransactionObject.data;
+
+    const nonce = await provider.getTransactionCount(account as string);
+    const predictedVaultAddress = ethers.utils.getContractAddress({
+      from: account as string,
+      nonce: nonce,
+    });
+    const zeroAddress = '0x0000000000000000000000000000000000000000';
+
+    //  2.Set the vault in the forwarder contract
+    const graveForwarderContract = new ethers.Contract(
+      networkConfig.universalGraveForwarder,
+      LSP1GraveForwarder.abi,
+      signer
+    );
+    const encodedSetGrave = graveForwarderContract.interface.encodeFunctionData(
```

Comment:
can this be done by calling normal function on contract or does it have to be low-level encoded for batching?

Reply:
I think it needs to be encoded, I'm not a fan either heheh